server/webdriver/tests/vis_scatter_test.py

```diff
@@ -0,0 +1,277 @@
+        chart_title.text,
+        "Population Without Health Insurance (2021) vs Female Population (2021)"
+    )
+    chart = self.driver.find_element(By.XPATH, '//*[@id="scatterplot"]')
+    circles = chart.find_elements(By.TAG_NAME, 'circle')
+    self.assertGreater(len(circles), 20)
+
+    # Click all the chart options and assert results are correct.
+    chart_option_inputs = self.driver.find_elements(
+        By.CSS_SELECTOR,
+        '.chart-footer-options .chart-option .form-check-input')
+    for input in chart_option_inputs:
+      input.click()
+      shared.wait_for_loading(self.driver)
+    y_axis_label = self.driver.find_element(By.CSS_SELECTOR,
+                                            '#scatterplot .y-axis-label')
+    self.assertTrue(
+        'Population Without Health Insurance (%)' in y_axis_label.text)
+    x_axis_label = self.driver.find_element(By.CSS_SELECTOR,
+                                            '#scatterplot .x-axis-label')
+    self.assertTrue('Female Population (%)' in x_axis_label.text)
+    chart = self.driver.find_element(By.XPATH, '//*[@id="scatterplot"]')
+    circles = chart.find_elements(By.TAG_NAME, 'circle')
+    self.assertGreater(len(circles), 20)
+    quadrant_lines = chart.find_elements(By.CSS_SELECTOR,
+                                         '#scatterplot .quadrant-line')
+    self.assertEqual(len(quadrant_lines), 2)
+    dot_labels = chart.find_element(By.CLASS_NAME, 'dot-label').find_elements(
+        By.TAG_NAME, 'text')
+    self.assertEqual(len(circles), len(dot_labels))
+
+    # Edit source and assert results are correct.
+    edit_source_button = self.driver.find_element(
+        By.CLASS_NAME, 'source-selector-open-modal-button')
+    edit_source_button.click()
+    element_present = EC.presence_of_element_located(
+        (By.CLASS_NAME, 'modal-body'))
+    WebDriverWait(self.driver, self.TIMEOUT_SEC).until(element_present)
+    source_option_sections = self.driver.find_elements(
+        By.CLASS_NAME, 'source-selector-options-section')
+    self.assertEqual(len(source_option_sections), 2)
+    # Open the selection section for each sv
+    self.driver.find_elements(By.CLASS_NAME,
+                              'source-selector-trigger')[0].click()
+    self.driver.find_elements(By.CLASS_NAME,
+                              'source-selector-trigger')[1].click()
+    # Update the source for the Count_Person_Female sv
+    element_present = EC.element_to_be_clickable(
+        (By.NAME, 'Count_Person_Female'))
+    WebDriverWait(self.driver, self.TIMEOUT_SEC).until(element_present)
+    source_options = self.driver.find_elements(By.NAME, 'Count_Person_Female')
+    self.assertGreater(len(source_options), 3)
+    source_options[3].click()
+    update_button = self.driver.find_element(By.CSS_SELECTOR,
+                                             '.modal-footer .btn')
+    update_button.click()
+    shared.wait_for_loading(self.driver)
+    # Check that results are correct
```

Comment:
for all of these, the lack of newline is because it's a subtask for the main task. In this section, the main task was "Edit source and assert results are correct" and this is just one of the steps for completing this. Let me know if you feel strongly about updating this and I can do this in a followup